lib/puppet/provider/package/puppet_gem.rb

```diff
@@ -4,8 +4,19 @@
 
   has_feature :versionable, :install_options, :uninstall_options
 
+  confine :true => Facter.value(:aio_agent_version)
```

Comment:
IIRC the `aio_agent_version` fact only exists in PE(?) which I think will make this provider no longer suitable for FOSS users?

Reply:
ah nvm, it looks like that is a core fact after all and it depends on the '/opt/puppetlabs/puppet/VERSION' file (or equivalent windows file), which should only be present when running from a puppet-agent package. Would you mind updating the commit message about why `aio_agent_version` was selected to detect puppet as a gem vs puppet-agent packages?